src/NzbDrone.Common/Http/HttpRequestBuilder.cs

```diff
@@ -253,6 +262,19 @@ public virtual HttpRequestBuilder SetHeader(string name, string value)
             return this;
         }
 
+        public virtual HttpRequestBuilder AddPrefixQueryParam(string key, object value, bool replace = false)
+        {
+            if (replace)
+            {
+                QueryParams.RemoveAll(v => v.Key == key);
+                SuffixQueryParams.RemoveAll(v => v.Key == key);
+            }
+
+            QueryParams.Insert(0, new KeyValuePair<string, string>(key, value.ToString()));
```

Comment:
In that case I think we should kill it, if your want your query string in a specific order then insert them in that order.